x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/TransactionFlyout/index.tsx

```diff
@@ -31,25 +41,58 @@ function getDiscoverQuery(id: string) {
 interface Props {
   onClose: () => void;
   transaction?: Transaction;
+  location: any; // TODO: import location type from react router or history types?
+  urlParams: IUrlParams;
+  waterfall: IWaterfall;
 }
 
-export function TransactionFlyout({ transaction, onClose }: Props) {
-  if (!transaction) {
+export function TransactionFlyout({
+  transaction: transactionGroup,
```

Comment:
Minor nit about names:

I think `transactionGroup` is a bit misleading, since we use that term to talk about groups of transaction (eg in aggregations transactions grouped by name is a transaction group).

Afaict this is a transaction document that contains the `transaction` property. So could be called `transactionDoc`. However, I'm not 100% sure we should rename it just yet, we have the `Transaction` interface, which in that case should be renamed `TransactionDoc`. 

Maybe I'm putting to much emphasis on naming.

Reply:
Ah, I'm getting confused between all of the different variations of transaction objects we have I think so I actually thought this was the group, but you're right it's the "transactionData" or "Doc" part that contains "transaction" etc ... I'll change this to transactionDoc for now because I want to destructure transaction from it later. We can rethink names later.